app/server.js

```diff
@@ -1,11 +1,42 @@
 require('dotenv').config();
+const util = require('util');
 const axios = require('axios');
 const config = require('./config');
 const { App } = require('./app');
 const { AuthProvider } = require('./authProvider');
 const { routes } = require('./routes');
 const { logger } = require('./logger');
 
+const setTimeoutPromise = util.promisify(setTimeout);
+
+const determineAppTypeToStart = async ({
+  canStartApp, appType, attempt, pollDuration,
+}) => {
+  if (!canStartApp) {
+    if (config.loginEnabled === 'false') {
+      return determineAppTypeToStart({
```

Comment:
arggghhhhh recursion 😭

Reply:
I dreamt this solution last night and was so excited to try it out this morning 😂